logictest: reset role before doing post-test checks

rafiss opened the pull request:
This makes sure we run the checks with the root user.

fixes https://github.com/cockroachdb/cockroach/issues/107561
Release note: None

rafiss committed:
logictest: reset role before doing post-test checks

This makes sure we run the checks with the root user.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/107626)
<!-- Reviewable:end -->

mgartner reviewed (approved):
Thanks! I had thought of doing something similar but wasn't sure if `setUser` was already supposed to be resetting the role. <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/107626#-:-NaIFNjg9JyJ2Vx5Bnti:b4b9fe5)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rafiss)



<!-- Sent from Reviewable.io -->

rafiss commented:
That is confusing - I think I'll rename `setUser` to be `setSessionUser` to clarify that it only changes the SessionUser

rafiss committed:
logictest: rename setUser to setSessionUser

This makes it clear that the "current user" can still be modified if one
executes SET ROLE commands in a logic test.

Release note: None

rafiss commented:
bors r+
